### ISSUE TITLE
SelectionList (keyboard handling)

### ISSUE THREAD

**Issue by tobich** [wix/stylable-components#85]
Implement item traversal with keyboard.

**Comment by alisey** [comment]
@tobich I had to deviate from the spec a little.

1) We need to know the value associated with each data item before we render them to DOM in order to determine if the item is selected/focused, and if it needs to be selectable via keyboard. But items, in general, are opaque objects. Even though `itemRenderer` puts a `data-value` attribute on the element in its `render` function, this function is not invoked when we call `itemNode = React.createElement(ItemRenderer, props, children)`. What we get at this point is not an entire subtree of the component, but simply a shallow object with the type of the component and its props. We have no clue about its internal structure. There are a few possible solutions:
a) Require the user to provide a function that takes an item and returns its value.
b) Add more structure to the `SelectionItem` interface. For example, require objects to have a `value` attribute: `string | {value: string, item: any}`
c) Require the user to put the `data-value` attribute on a JSX element returned by `itemRenderer`. To make this possible itemRenderer needs to become a function, `renderItem(item): React.Node`. Then we can read `data-value` from the returned node. I went with this approach, because it unifies the interface between dataSource and direct children of `SelectionList`. We need to read the `data-value` attribute from children anyway. Another little benefit is that renderItem can directly return a `<div>` element without having to wrap it into a component.

2) Since the user of the library is not required to use stylable, we cannot simply add cssState to each node returned by `renderItem`. Maybe the item renderer even wants to return completely different components depending on whether the item is selected or not. So ideally we would pass `selected` and `focused` as arguments to `renderItem`. But what I'm currently doing is adding `selected` and `focused` as props to the element returned by `renderItem`. This certainly is not ideal at least from discoverability standpoint, and I want to discuss with you how we can improve this.

3) The native `<select>` element allows an item to be both selected and disabled. If we want to support this feature, then SelectionList needs to be aware about `disabled` status of the items. I can imagine users trying to do: `<div data-value="1" data-disabled>One</div>`.
